doc/devdocs/modules/launcher/plugins/history.md

```diff
@@ -0,0 +1,103 @@
+We then also add the "Remove this from history" context menu action.
+
+## Results score
+When the plugin is used with the activation command, the scores are configured so the results show with the more recently selected items at the top.
+If the history results are shown in the global results, the scores are not modified from that the original plugin set.
+
+## Old Data
+Items selected before this plugin was created will not show in the history because they don't contain enough data.
+
+## Important for developers
+
+### Important plugin values (meta-data)
+
+| Name            | Value                                                |
+| --------------- | ---------------------------------------------------- |
+| ActionKeyword   | `!!`                                                  |
+| ExecuteFileName | `Microsoft.PowerToys.Run.Plugin.History.dll` |
+| ID              | `C88512156BB74580AADF7252E130BA8D`                   |
+
+### Interfaces used by this plugin
+
+The plugin uses only these interfaces (all inside the `Main.cs`):
+
+* `Wox.Plugin.IPlugin`
+* `Wox.Plugin.IContextMenu`
+* `Wox.Plugin.IPluginI18n`
+
+### Program files
+
+| File                                  | Content                                                                 |
+| ------------------------------------- | ----------------------------------------------------------------------- |
+| `Images\history.dark.png`             | Symbol for the results for the dark theme                               |
+| `Images\history.light.png`            | Symbol for the results for the light theme                              |
+| `Properties\Resources.Designer.resx`  | File that contain all translatable keys                                 |
+| `Properties\Resources.resx`           | File that contains all translatable strings in the neutral language      |
+| `Main.cs`                             | Main class, the only place that implements the WOX interfaces            |
+| `ErrorHandler.cs`                     | Class to build error result on plugin failure                           |
+| `plugin.json`                         | All meta-data for this plugin                                           |
+
+### Important project values (*.csproj)
+
+| Name            | Value                                             |
+| --------------- | ------------------------------------------------- |
+| TargetFramework | `net6.0-windows10.0.19041.0`                      |
+
+### Project dependencies
+
+#### Projects
+
+* `Wox.Infrastructure`
+* `Wox.Plugin`
+* `PowerToys.PowerLauncher`
+
+
+#### Build Dependency
+Access to PluginManager was needed to make this plugin work. Because of this a reference to PowerToys.PowerLauncher was needed.
+Since History Plugin needs a reference to PowerToys.PowerLauncher, PowerToys.PowerLauncher cannot depend on History Plugin (else a circular reference would exist).
```

Comment:
```suggestion
Since History Plugin needs a reference to PowerToys.PowerLauncher, it can not be set as a dependecy reference in PowerToys.PowerLauncher  project (else a circular reference would exist).
```